lightningd/log.c

```diff
@@ -289,6 +295,12 @@ void logv_add(struct log *log, const char *fmt, va_list ap)
 	list_del_from(&log->lr->log, &l->list);
 
 	tal_append_vfmt(&l->log, fmt, ap);
+
+	/* Sanitize any non-printable characters, and replace with '?' */
+	for (size_t i=0; i<strlen(l->log); i++)
+		if (l->log[i] < ' ' || l->log[i] >= 0x7f)
```

Comment:
Should we be checking == 0x7f so we let through UTF-8?  That would seem sensible...

Reply:
As much as I'd like to have UTF-8 support in here, that'd mean we now need to carry context and have multi-char replacements in some cases, e.g., there are multi-byte control characters again. I'm afraid that we probably won't get away without fully parsing UTF-8 at some point, so I'd prefer keeping this ASCII only for now, and have an issue open to support full UTF-8 with corresponding escaping.